### PR TITLE
docs: explain thinking token budgets

### PR DESCRIPTION
**Pull request by biswapanda**
## Summary

- Document the public root-level `thinking_token_budget` request field.
- Explain the vLLM sampling-parameter mapping and SGLang strict-thinking setup.
- Describe SGLang restrictions for custom logit processors and unsupported raw-tokenizer or native-passthrough paths.

## Validation

- `python3 docs/fern/scripts/docs_lint.py --scan docs`
- `pre-commit run --files docs/fern/pages/use-cases/tool-calling-and-reasoning/reasoning-parsing.md`
- `fern check` after generating local API references: passed with three pre-existing warnings.
- `fern docs broken-links` was blocked by an unrelated existing broken DeepSeek recipe link.

## Stack

- Base: #15121
- Depends on: #15120

### CODE REVIEW

File: docs/fern/pages/use-cases/tool-calling-and-reasoning/reasoning-parsing.md [hunk 1]
```diff
@@ -169,6 +169,58 @@ deployment default.
...
+The budget must be a non-negative integer. A value of `0` prevents additional reasoning-token generation while leaving the request valid.
+
+### vLLM
+
+Dynamo forwards `thinking_token_budget` to vLLM's `SamplingParams.thinking_token_budget` when the installed vLLM version exposes that field. Configure reasoning for the model as usual, then send the request above through the Dynamo frontend.
+
+```bash
+python -m dynamo.vllm \
+  --model Qwen/Qwen3-0.6B \
+  --dyn-reasoning-parser qwen3
```
[the remark this answers is not in the document]
Comment (biswapanda) [reply docs/fern/pages/use-cases/tool-calling-and-reasoning/reasoning-parsing.md:200]: Added vLLM native --reasoning-parser qwen3 while retaining Dynamo response parsing.

File: docs/fern/pages/use-cases/tool-calling-and-reasoning/reasoning-parsing.md [hunk 2]
```diff
@@ -169,6 +169,59 @@ deployment default.
...
+      "role": "user",
+      "content": "Return a short answer for this synthetic prompt."
+    }
+  ],
+  "max_completion_tokens": 128,
+  "thinking_token_budget": 32
+}
+```
+
+The budget must be a non-negative integer. A value of `0` prevents additional reasoning-token generation while leaving the request valid.
```
[the remark this answers is not in the document]
Comment (biswapanda) [reply docs/fern/pages/use-cases/tool-calling-and-reasoning/reasoning-parsing.md]: fixed in b75e099b5f

[the remark this answers is not in the document]
Comment (biswapanda) [reply docs/fern/pages/use-cases/tool-calling-and-reasoning/reasoning-parsing.md]: fixed in b75e099b5f

File: docs/fern/pages/use-cases/tool-calling-and-reasoning/reasoning-parsing.md [hunk 3]
```diff
@@ -169,6 +169,59 @@ deployment default.
...
+  --model Qwen/Qwen3-0.6B \
+  --reasoning-parser qwen3 \
+  --dyn-reasoning-parser qwen3
+```
+
+Use a vLLM release that provides the built-in thinking-budget logits processor. If the active vLLM `SamplingParams` does not expose `thinking_token_budget`, Dynamo cannot apply a per-request budget.
+
+### SGLang
+
+SGLang requires strict thinking, an SGLang reasoning parser with an active reasoning-token filter, an initialized tokenizer, and a grammar backend. Run the SGLang chat processor in the frontend so Dynamo can preserve the canonical preprocessed request state.
```
[the remark this answers is not in the document]
Comment (biswapanda) [reply docs/fern/pages/use-cases/tool-calling-and-reasoning/reasoning-parsing.md]: fixed in 14975c244d

File: docs/fern/pages/use-cases/tool-calling-and-reasoning/reasoning-parsing.md [hunk 4]
```diff
@@ -169,6 +169,59 @@ deployment default.
...
+  --model Qwen/Qwen3-0.6B \
+  --reasoning-parser qwen3 \
+  --dyn-reasoning-parser qwen3
+```
+
+Use a vLLM release that provides the built-in thinking-budget logits processor. If the active vLLM `SamplingParams` does not expose `thinking_token_budget`, Dynamo cannot apply a per-request budget.
+
+### SGLang
+
+SGLang requires strict thinking, an SGLang reasoning parser with an active reasoning-token filter, an initialized tokenizer, and a grammar backend. Budgeted requests must also enable reasoning through the model's request-level control. For templates that use `enable_thinking`, send `"chat_template_kwargs": {"enable_thinking": true}`; do not disable thinking on a request with `thinking_token_budget`. Run the SGLang chat processor in the frontend so Dynamo can preserve the canonical preprocessed request state.
```
[the remark this answers is not in the document]
Comment (biswapanda) [reply docs/fern/pages/use-cases/tool-calling-and-reasoning/reasoning-parsing.md]: fixed in 8655dec7b4
